### DISCUSSION TITLE
Parquet set row_group size

### DISCUSSION THREAD

**Discussion by darthf1** [duckdb/duckdb#3054] [Q&A] [answered]
Hi!

I'm trying writing the result of a query to a parquet file, using the following statement (taken from the docs):
```
COPY (SELECT * FROM tbl) TO 'result-snappy.parquet' (FORMAT 'parquet');
```

Right now I'm seeing the error:

```
RuntimeError: Out of Memory Error: failed to pin block of size 262205
```

I don't want to touch the available memory on the machine, so I'm guessing this could be fixed by lowering the `row_group_size`? Is it possbile to set a `row_group_size` somewhere in the statement above?

I guess now its set to [`100000`](https://github.com/duckdb/duckdb/blob/59838eae344ca54e243a08ecc90e1c38ae695aa3/extension/parquet/parquet-extension.cpp#L398)?

**Comment by Alex-Monahan** [comment]
Hello!

As a workaround, have you tried setting the maximum memory that DuckDB will use? After that point it should use a temporary file as a buffer. Try using 80% or less of your available RAM.

```sql
-- set the memory limit
PRAGMA memory_limit='1GB';
```
[PRAGMA Docs](https://duckdb.org/docs/sql/pragmas)

**Comment by Mytherin** [comment, marked as answer]
You can set the row group size using the `ROW_GROUP_SIZE` parameter:

```sql
COPY tbl TO 'file.parquet' (FORMAT 'PARQUET', ROW_GROUP_SIZE 25000);
```
